CONTRIBUTING.md

```diff
@@ -0,0 +1,98 @@
+* Everything submitted as a PR should have at least one test, the only exception being non-code items.
+
+## Overview of the Bridge
+
+The bridge runs as a standalone server that connects to both the Discord API
+network and a local Matrix homeserver over the [application service
+protocol](https://matrix.org/docs/spec/application_service/unstable.html).
+Primarily it syncs events and users from Matrix to Discord and vice versa.
+
+While the bridge is constantly evolving and we can't keep this section updated
+with each component, we follow the principle of handler and processor classes
+and each part of the functionality of the bridge will be in a seperate class.
+For example, the processing of Matrix events destined for Discord are handled
+inside the ``MatrixEventProcessor`` class.
+
+## Setting up
+
+(This section needs to be written in detail)
+
+* You will need to [setup the bridge](https://github.com/Half-Shot/matrix-appservice-discord/tree/develop#setup-the-bridge) similarly to how we describe,
+  but you should setup a homeserver locally on your development machine. We would recommend [Synapse](https://github.com/matrix-org/synapse#id11).
+
+## Writing an issue
+
+When writing an issue, please be as verbose as you can. Remember the issue is
+there to either document a feature request, or report a bug so it can be fixed.
+The issue board is NOT there to complain about a broken or missing feature.
+
+We leave it to the author's disgression to decide what to include rather than
+provide a template, but good items are:
+ * Shorter titles are better than long rambling ones, but please don't make it too vague.
+    * A good example is "Ability to bridge an existing matrix room to a discord channel"
+ * A brief description of the problem.
+ * Relevant logging from Synapse/your homeserver AND the bridge.
+    * If you are a user of another person's bridge, please can you give the name of that person or homeserver.
+    * The more verbose, the better but please don't include sensitive details like access tokens.
+      If you must, please PM Half-Shot discreetly.
+ * A screenshot.
+
+We will assign each issue a tag, which will allow us to categorise the problems.
+
+Demands from users to fix issues on any kind of time scale, without a very good
+reason is not tolerated and will not help your issue to get fixed quicker. The
+Code of Conduct linked above also applies to issues and we will take any
+complaints very seriously.
+
+Please bear with us as we get to your issue, the project runs without any
+external funding or support except from the work of the community and as such
+offer no on-demand support or feature development.
+
+
+## PR Process
+
+We've tried our best to keep the PR process relatively simple:
+
+* Create a new branch off develop and name it ``yourusernameoridentifer/featurename``.
+    * This can be done with ``git checkout develop`` followed by ``git checkout -b yourusernameoridentifer/featurename``.
```

Comment:
Is the `yournameoridentifier` part necessary? The author will most likely work on his own forked version of the repo anyway.

Reply:
how about having this guideline only for branches in the official repo itself?